agent/cache-types/streaming_health_services.go

```diff
@@ -206,6 +208,21 @@ func (noopFilterEvaluator) Evaluate(_ interface{}) (bool, error) {
 	return true, nil
 }
 
+// cachedHealResultSorter sorts the results to match memdb semantics
+// Sort results by Node.Node, if 2 instances match, order by Service.ID
+// Will allow result to be stable sorted and match queries without cache
+func cachedHealResultSorter(serviceNodes *structs.IndexedCheckServiceNodes) {
+	sort.SliceStable(serviceNodes.Nodes, func(i, j int) bool {
+		left := serviceNodes.Nodes[i]
+		right := serviceNodes.Nodes[j]
+		res := strings.Compare(left.Node.Node, right.Node.Node)
+		if res != 0 {
+			return res < 0
+		}
+		return strings.Compare(left.Service.ID, right.Service.ID) < 0
+	})
```

Comment:
According to https://golang.org/pkg/strings/#Compare, "strings.Compare is included only for symmetry with package bytes. It is usually clearer and always faster to use the built-in string comparison operators ==, <, >, and so on."

So I guess this should use `<` instead of `strings.Compare`.

```go
left := serviceNodes.Nodes[i]
right := serviceNodes.Nodes[j]
if left == right {
    return left.Service.ID < right.Service.ID
}
return left.Node.Node < right.Node.Node
```